File: BinaryTrees_Repeat/PathSum III.cpp


/*
class Solution{
public:
    void helper(TreeNode* root, long long sum, int &count){
        if(root == NULL) return;
        if((long long)(root->val) == sum){
            cout++;
        }
        helper(root->left, sum-(long long)(root->val),count);
        helper(root->right,sum-(long long)(root->val),count);
    }
    int pathSum(Treenode* root, int targetSum){
        // to add the sums of every nodepath
        if(root == NULL) return 0;
        int count = 0;
        helper(root, (long long)(targetSum),count);
        count += (pathSum(root->left, targetSum) + pathSum(root->right,targetSum));
        return count;
    }
}
*/

File: BinaryTrees_Repeat/Path Sum III.java
/*

class Solution{
    public int noOfPaths(TreeNode root, long targetSum){
        if(root == null) return 0;
        int count = 0;
        if((long)(root.val) == targetSum) count++;
        return count + noOfPaths(root.left, targetSum - (long)(root.val)) + noOfPaths(root.right,targetSum - (long)(root.val));
        
    }
    public int PathSum(treeNode root, int targetSum){
        if(root == null) return 0;
        int count = noOfPaths(root,(long)(targetSum));
        count += (pathSum(root.left,targetSum) + pathsum(root.right,targetSum));
        return count;
    }
}

*/